docs/how_to_use_csctl_plugin_openstack.md

```diff
@@ -0,0 +1,61 @@
+# Using the csctl plugin for OpenStack
+
+## What does the csctl plugin for OpenStack do?
+
+As a user, you can create clusters based on Cluster Stacks with the help of the Cluster Stack Operators (CSO and CSPO). The operators need certain files, such as those required to apply the necessary Helm charts and to obtain information about the versions in the cluster stack.
+
+To avoid generating these files manually, you can use [CSCTL](https://github.com/SovereignCloudStack/csctl). In the case of provider-specific Cluster Stacks, the `CSCTL` tool invokes the provider-specific CSCTL plugin. Therefore, the CSCTL plugin for OpenStack is essential if the user intends to build, upload node images to an S3 bucket, and then import them into Glance.
```

Comment:
I think there is no example missing. You already have one in `example` folder. If anyone wants to know how csctl invokes plugins, you need to go to the `csctl` repository because all plugins are invoked the same way.

Reply:
I don't know where it is written but, what I see missing is following:
1. plugin is invoked when `Config.Provider.Config` is not empty - we support `method: get/build`
2. `config.provider.type` have to be `openstack`
3. there has to be folder `node-images` with `config.yaml`

Reply:
> 1. plugin is invoked when `Config.Provider.Config` is not empty - we support `method: get/build`

This is general information on how `csctl` invokes all plugins, so from my point of view, this should be in `csctl` docs rather. We name them exactly what we support in this section `Different methods of csctl plugin for OpenStack`. Also from the example of the `csctl.yaml` you see how the get method is specified.

> 2. `config.provider.type` have to be `openstack`

Hmm, I think you do not need to say that exactly somewhere, as this is a plugin for OpenStack, we also provide an example file, where this is specified, and nowhere is it said in the plugin documentation that it can be something else.

> 3. there has to be folder `node-images` with `config.yaml`

Yes, I agree that defining precisely that the `node-images` folder must always exist and contain `config.yaml` for both methods can be helpful for newcomers.